resource_api_endpoints.yaml

```diff
@@ -0,0 +1,255 @@
+        description: Create new resource
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Resource'
+        required: true
+      responses:
+        '201':
+          description: Resource created successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Resource'    
+        '400':
+          description: Invalid input
+  /resources/{id}:
+    get:
+      tags:
+        - resource
+      summary: Get resource by id
+      parameters:
+        - name: id
+          in: path
+          description: ID of resource to return
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+       '200':
+         description: successful operation
+         content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Resource'
+       '404':
+         description: resource not found
+    put:
+      tags:
+      - resource
+      summary: Update a resource
+      parameters:
+        - name: id
+          in: path
+          description: ID of resource to edit
+          required: true
+          schema:
+            type: integer
+            format: int64
+      requestBody:
+        description: updated resource
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Resource'
+      responses:
+        '200':
+          description: Resource updated successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Resource'
+        '404':
+          description: Resource not found
+    delete:
+      tags:
+      - resource
+      summary: Delete a resource
+      parameters:
+        - name: id
+          in: path
+          description: ID of resource to delete
+          required: true
+          schema:
+            type: integer
+      responses:
+        '204':
+          description: Resource deleted successfully
+        '404':
+          description: Resource not found
+    
+  /resources/{id}/views:
+    get:
+      tags:
+        - resource_views
+      summary: get view count for a resource
+      parameters:
+        - name: id
+          in: path
+          description: ID of resource
+          required: true
+          schema:
+            type: integer
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                  type: object
+                  properties:
+                     resource_id:
+                       type: integer
+                     view_count:
+                       type: integer
+                       example: 1000
+        '404':
+          description: resource not found
+  /resources/{id}/view:
+    post:
```

Comment:
I don't think a post method is required
Since with each get request the user will do, the backend service will handle incrementing the value